mx.py

```diff
@@ -3972,7 +3972,7 @@ def getSnapshot(self, groupId, artifactId, version):
         try:
             metadataFile = urllib2.urlopen(metadataUrl)
         except urllib2.URLError as e:
-            if e.code == 404:
+            if isinstance(e, urllib2.HTTPError) and e.code == 404:
                 return None
             abort('Error while retreiving snappshot for {}:{}:{}: {}'.format(groupId, artifactId, version, str(e)))
```

Comment:
while you're at it, you could fix the typos here, too: s/retreiving/retrieving/ s/snappshot/snapshot/